### PR TITLE
ARROW-622 [Python] deprecate timestamps_to_ms in .from_pandas()

### PR DESCRIPTION
**Pull request by jreback**
xref https://github.com/pandas-dev/pandas/issues/17438

this was not fully resolved in https://github.com/apache/arrow/pull/944

### CODE REVIEW

File: python/pyarrow/table.pxi [hunk 1]
```diff
@@ -701,18 +701,25 @@ cdef class Table:
         return result
 
     @classmethod
-    def from_pandas(cls, df, bint timestamps_to_ms=False,
+    def from_pandas(cls, df,
+                    object coerce_timestamps=False, object timestamps_to_ms=None,
```
Comment (jreback) [inline python/pyarrow/table.pxi]: note that the signature is slightly changed here; I could add the new argument at the end if you prefer.

**Comment by jreback** [comment]
green other that a stalled build: https://travis-ci.org/apache/arrow/jobs/272034734

File: python/pyarrow/array.pxi [hunk 2]
```diff
@@ -116,10 +117,16 @@ cdef class Array:
...
 
-        timestamps_to_ms : bool, optional
-            Convert datetime columns to ms resolution. This is needed for
-            compatibility with other functionality like Parquet I/O which
-            only supports milliseconds.
+        coerce_timestamps : {'ms', 'us', False}, default False
+            Convert datetime columns to the indicated resolution. If False
+            preserve the existing dtype.
+            This is needed for compability with other functionality like
+            Parquet I/O which only supports milliseconds / microseconds.
```
Comment (wesm) [inline python/pyarrow/array.pxi]: I don't think this is the right fix. The coercion happens in `parquet.write_table`

**Review by wesm — COMMENTED** [review]
I'm -1 on this change unless there is definitely no other way. Can this be handled by passing `coerce_timestamps` to `parquet.write_table`?

**Comment by jreback** [comment]
ok this is more limited in scope, removing test warnings for using ``timestamp_to_ms`` in ``.from_pandas()``.

I'll fix the deprecation issue in .to_parquet on the pandas side itself.

**Comment by wesm** [comment]
LGTM. The build is failing due to a Sphinx warning, can you fix that?

**Comment by wesm** [comment]
+1. Build failed due to stalled macos image and maven central flakiness

**Pull request closed without merging**